### ISSUE TITLE
User update icon (yellow dot) is no longer displayed

### ISSUE THREAD

**Issue by sjoubert** [Kitware/CDash#1482]
# Bug report

### Expected Behavior

Display the expected indicator if I'm logged in and I have updates on a build.

### Actual Behavior

Nothing is displayed, ever.

### CDash Version

v3.1.0

### Additional Information

It seems the relevant query is injected here: https://github.com/Kitware/CDash/blob/v3.1.0/app/cdash/app/Controller/Api/Index.php#L119 but it seems the session entry write was removed in 86c06487bf8caabca4157299c26f5e2b10664082
For what it's worth at a quick glance there's couple other reads of _SESSION but no obvious write of information for them.

**Comment by sjoubert** [comment]
This patch (on v3.1.0) seems to fix the issue.
```patch
diff --git a/app/cdash/app/Controller/Api/Index.php b/app/cdash/app/Controller/Api/Index.php
index 837357fcd..501582430 100644
--- a/app/cdash/app/Controller/Api/Index.php
+++ b/app/cdash/app/Controller/Api/Index.php
@@ -116,13 +116,13 @@ class Index extends ResultsApi
 
         // If the user is logged in we display if the build has some changes for them.
         $userupdatesql = '';
-        if (isset($_SESSION['cdash']) && array_key_exists('loginid', $_SESSION['cdash'])) {
+        if (\Auth::check()) {
             $userupdatesql = "(SELECT count(updatefile.updateid) FROM updatefile,build2update,user2project,
                 user2repository
                     WHERE build2update.buildid=b.id
                     AND build2update.updateid=updatefile.updateid
                     AND user2project.projectid=b.projectid
-                    AND user2project.userid='" . $_SESSION['cdash']['loginid'] . "'
+                    AND user2project.userid='" . \Auth::id() . "'
                     AND user2repository.userid=user2project.userid
                     AND (user2repository.projectid=0 OR user2repository.projectid=b.projectid)
                     AND user2repository.credential=updatefile.author) AS userupdates,";
```

**Comment by sjoubert** [comment]
Oh, this seems to have been fixed since v3.1.0, in fc3a629e0a6425905b78c83c1f2120d531ac1786
I can't confirm this is fixed in the current code, but feel free to close this issue if you do.
